Common/Symbol.cs

```diff
@@ -307,6 +307,10 @@ public Symbol(SecurityIdentifier sid, string value)
                 throw new ArgumentNullException(nameof(value));
             }
             ID = sid;
+            if (ID.HasUnderlying)
+            {
+                Underlying = new Symbol(ID.Underlying, ID.Underlying.Symbol);
```

Comment:
Nice find!
This is exposing an interesting 'information duplication' case, the fact that the `Symbol Symbol.Underlying` isn't directly attached to it's `SecurityIdentifier ID.Underlying`, which means they could be different. With this change we should be covered since Symbol only has two constructors.